cotainr/pack.py

```diff
@@ -281,6 +286,35 @@ def _display_miniforge_license_for_acceptance(self, *, installer_path):
                 "No license seems to be displayed by the Miniforge installer."
             )
 
+    @staticmethod
+    def _get_install_script(architecture):
+        """
+        Determine the Miniforge installer to be downloaded based on system architecture.
+
+        Always downloads a Linux version as the container is expected to always be Linux.
+
+        Parameters
+        ----------
+        architecture : str
+            The container architecture as returned by "uname -m".
+
+        Raises
+        ------
+        NotImplementedError
+            If the container sandbox architecture is not supported.
+        """
+        if architecture in ("arm64", "aarch64"):
+            install_script = "Miniforge3-Linux-aarch64.sh"
+        elif architecture == "x86_64":
+            install_script = "Miniforge3-Linux-x86_64.sh"
+        else:
+            raise NotImplementedError(
+                "Cotainr's CondaInstall only supports x86_64 and arm64/aarch64. "
+                f'Cotainr got "{architecture=}" for your container'
+            )
+
+        return install_script
```

Comment:
This would be simpler to read with early exits, IMO:

```suggestion
        if architecture in ("arm64", "aarch64"):
            return "Miniforge3-Linux-aarch64.sh"
        
        if architecture == "x86_64":
            return "Miniforge3-Linux-x86_64.sh"
        
        raise NotImplementedError(
            "Cotainr's CondaInstall only supports x86_64 and arm64/aarch64. "
            f'Cotainr got "{architecture=}" for your container'
        )
```